File: server/php/src/Application/DTOs/HostedCheckout/RequestDto.php
<?php

namespace OnlinePayments\ExampleApp\Application\DTOs\HostedCheckout;

use OnlinePayments\ExampleApp\Application\Domain\Enums\Common\Currency;
use OnlinePayments\ExampleApp\Application\Domain\Enums\Common\Language;
use OnlinePayments\ExampleApp\Application\DTOs\Common\AddressDto;

class RequestDto
{
    public ?int $amount;
    public ?Currency $currency;
    public ?Language $language;
    public ?AddressDto $billingAddress;
    public ?AddressDto $shippingAddress;
    public ?string $redirectUrl;

    public function __construct(
        ?int $amount,
        ?Currency $currency,
        ?Language $language = null,
        ?AddressDto $billingAddress = null,
        ?AddressDto $shippingAddress = null,
        ?string $redirectUrl = null
    ) {
        $this->amount = $amount;
        $this->currency = $currency;
        $this->language = $language;
        $this->billingAddress = $billingAddress;
        $this->shippingAddress = $shippingAddress;
        $this->redirectUrl = $redirectUrl;
    }
}

File: server/python/app/application/dtos/hosted_checkout/request_dto.py
from dataclasses import dataclass
from typing import Optional
from app.application.domain.enums.common.currency import Currency
from app.application.domain.enums.common.language import Language
from app.application.dtos.common.address_dto import AddressDto

@dataclass
class CreateHostedCheckoutRequestDto:
    amount: Optional[int] = None
    currency: Optional[Currency] = None
    language: Optional[Language] = None
    billing_address: Optional[AddressDto] = None
    shipping_address: Optional[AddressDto] = None
    redirect_url: Optional[str] = None